### PR TITLE
Fixed betting tests

### PR DESCRIPTION
**Pull request by srpatel19590**
The primary issue causing the betting test fail is due to the new `insert` function added in `class primary_index` file `index.hpp`. 

```
virtual const object& insert( object&& obj )override
{
	 const auto& res = DerivedIndex::insert(std::move(obj));
	 for( const auto& item : _sindex )
		item->object_inserted( res );
	 on_add(res);
	 return res;
}
```

This function was added to fix SON tests (SON-142) #176 . The fix provided was correct for SON test. But on the other hand betting tests are failing with this change.

The reason behind it is the `secondary_index`. 

```
for( const auto& item : _sindex )
    item->object_inserted( res );
```
		
As shown in the above snippet, It executes the `object_inserted` function of `secondary_index`. There are various `secondary_index` that have been added for betting functionality in `bookie_plugin`. Let's take the example of `bet_object`.

For the `persistent_bet_object_helper` is added as `secondary_index` of `bet_object_index`. so `object_inserted` function of `persistent_bet_object_helper` class will be called from `insert` function of `primary_index` class and try to insert object in the db which is already available. This gives `insert_result.second: Could not create object! Most likely a uniqueness constraint is violated.` error.

I also tried implementing `object_removed` function in `persistent_bet_object_helper` class assuming it will be called when 'remove' of 'primary_index' will be called to remove the `persistent_bet_object_helper` object but that leads to certain other issues in betting tests.

I believe the `bookie_plugin` functionality was implemented assuming no undo required.

**Solution:**
To fix the betting tests we need to remove the `insert` function added in `class primary_index` file `index.hpp`.
To fix the SON tests we need to remove `secondary_index` of `son_object`. So `add_index< primary_index<son_index, 8> >(); ` is changed to `add_index< primary_index<son_index> >();`
 
With these two changes, all test cases are working fine.

### CODE REVIEW

File: libraries/db/include/graphene/db/index.hpp [hunk 1]
```diff
@@ -402,14 +402,17 @@ namespace graphene { namespace db {
...
 
-         virtual const object& insert( object&& obj )override
-         {
-             const auto& res = DerivedIndex::insert(std::move(obj));
-             for( const auto& item : _sindex )
-                item->object_inserted( res );
-             on_add(res);
-             return res;
-         }
+         // Note: Implementing insert function here will break the
```
Comment (oxarbitrage) [inline libraries/db/include/graphene/db/index.hpp]: Please remove commented code. If we need to know how this was done we can refer to specific commit, pull request, etc. No need to keep it.

File: libraries/chain/db_init.cpp [hunk 2]
```diff
@@ -263,7 +263,10 @@ void database::initialize_indexes()
    acnt_index->add_secondary_index<account_referrer_index>();
 
    add_index< primary_index<committee_member_index, 8> >(); // 256 members per chunk
-   add_index< primary_index<son_index, 8> >(); // 256 sons per chunk
+
+   // Note: Do not create secondary_index if you will need to create an 
+   // operation to remove associated object.
```
Comment (oxarbitrage) [inline libraries/chain/db_init.cpp]: Remove comment.

**Comment by oxarbitrage** [comment]
Looks good, i didn't tested, building now.

**Comment by bobinson** [comment]
> graphene::chain::vesting_policy_context&) const: Assertion `withdrawn_already >= 0' failed.
unknown location(0): fatal error: in "son_operation_tests/delete_son_test": signal: SIGABRT (application abort requested)


@srpatel19590  : what about this error in the tests ?

**Comment by oxarbitrage** [comment]
Error not happening here with this PR:

```
oxarbitrage@oxarbitrage:~/CLionProjects/pp217/peerplays$ ./tests/chain_test -t son_operation_tests/delete_son_test
Random number generator seeded to 1573492877
GRAPHENE_TESTING_GENESIS_TIMESTAMP is 1431700002
Running 1 test case...
1277975ms th_a       db_management.cpp:203         open                 ] Wiping object_database due to missing or wrong version
1277975ms th_a       object_database.cpp:93        wipe                 ] Wiping object database...
1277975ms th_a       object_database.cpp:95        wipe                 ] Done wiping object databse.
1277975ms th_a       object_database.cpp:106       open                 ] Opening object database from /tmp/graphene-tmp/d581-3735-ccab-c475 ...
1277975ms th_a       object_database.cpp:111       open                 ] Done opening object database.
1277997ms th_a       bookie_plugin.cpp:469         plugin_initialize    ] bookie plugin: plugin_startup() begin
1277997ms th_a       db_management.cpp:278         force_slow_replays   ] enabling slow replays
1277997ms th_a       bookie_plugin.cpp:502         plugin_initialize    ] bookie plugin: plugin_startup() end
1277997ms th_a       bookie_plugin.cpp:507         plugin_startup       ] bookie plugin: plugin_startup()
1277998ms th_a       db_maint.cpp:1218             process_dividend_ass ] In process_dividend_assets time 2015-05-15T14:26:45
1277998ms th_a       db_maint.cpp:892              schedule_pending_div ] Processing dividend payments for dividend holder asset type TEST at time 2015-05-15T14:26:45
1278010ms th_a       db_maint.cpp:948              schedule_pending_div ] Current balances in distribution account: 0, Previous balances: 0
1278016ms th_a       db_maint.cpp:1218             process_dividend_ass ] In process_dividend_assets time 2019-11-11T16:21:18
1278016ms th_a       db_maint.cpp:892              schedule_pending_div ] Processing dividend payments for dividend holder asset type TEST at time 2019-11-11T16:21:18
1278037ms th_a       db_maint.cpp:948              schedule_pending_div ] Current balances in distribution account: 0, Previous balances: 0
1278041ms th_a       db_maint.cpp:1243             process_dividend_ass ] Dividend payout time has arrived for asset TEST
1278041ms th_a       db_maint.cpp:1370             operator()           ] dividend_data_obj.last_scheduled_payout_time: 2015-05-16T14:26:42 dividend_data_obj.last_payout_time: 2019-11-11T16:21:18 dividend_data_obj.options.next_payout_time: 2019-11-21T14:26:42 
1278043ms th_a       db_maint.cpp:1218             process_dividend_ass ] In process_dividend_assets time 2019-11-13T16:21:42
1278043ms th_a       db_maint.cpp:892              schedule_pending_div ] Processing dividend payments for dividend holder asset type TEST at time 2019-11-13T16:21:42
1278064ms th_a       db_maint.cpp:948              schedule_pending_div ] Current balances in distribution account: 0, Previous balances: 0
1278069ms th_a       db_management.cpp:256         close                ] Database close unexpected exception: {"code":10,"name":"assert_exception","message":"Assert Exception","stack":[{"context":{"level":"error","file":"fork_database.cpp","line":41,"method":"pop_block","hostname":"","thread_name":"th_a","timestamp":"2019-11-11T17:21:18"},"format":"_head: no blocks to pop","data":{}},{"context":{"level":"warn","file":"db_block.cpp","line":553,"method":"pop_block","hostname":"","thread_name":"th_a","timestamp":"2019-11-11T17:21:18"},"format":"","data":{}}]}

*** No errors detected
oxarbitrage@oxarbitrage:~/CLionProjects/pp217/peerplays$ 
```

`betting_tests` also passing.

**Comment by bobinson** [comment]
Tried the following tests manually and like @srpatel19590 and @oxarbitrage mentioned, they are working fine.

```
    ./tests/betting_test --log_level=message
    ./tests/chain_test --log_level=message
    ./tests/cli_test --log_level=message
```

**Pull request merged**